core/trino-main/src/test/java/io/trino/sql/planner/TestDomainTranslator.java

```diff
@@ -552,6 +554,39 @@ public void testFromOrPredicate()
         assertEquals(result.getTupleDomain(), withColumnDomains(ImmutableMap.of(C_BIGINT, Domain.create(ValueSet.ofRanges(Range.range(BIGINT, 1L, false, 5L, false)), false))));
     }
 
+    @Test
+    public void testFromSingleBooleanReference()
+    {
+        Expression originalPredicate = C_BOOLEAN.toSymbolReference();
+        ExtractionResult result = fromPredicate(originalPredicate);
+        assertEquals(result.getTupleDomain(), withColumnDomains(ImmutableMap.of(C_BOOLEAN,
+                Domain.create(ValueSet.ofRanges(Range.equal(BOOLEAN, true)), false))));
+        assertEquals(result.getRemainingExpression(), TRUE_LITERAL);
+
+        originalPredicate = not(C_BOOLEAN.toSymbolReference());
+        result = fromPredicate(originalPredicate);
+        assertEquals(result.getTupleDomain(), withColumnDomains(ImmutableMap.of(C_BOOLEAN,
+                Domain.create(ValueSet.ofRanges(Range.equal(BOOLEAN, true)).complement(), false))));
```

Comment:
is it just `Range.equal(BOOLEAN, false))`?

Reply:
If I use Range.equal(BOOLEAN, false)), i. e. 
withColumnDomains(ImmutableMap.of(C_BOOLEAN,
                Domain.create(ValueSet.ofRanges(Range.equal(BOOLEAN, false)), false)))) 
than test fails because in this case the domains = Optional[{c_boolean=[ SortedRangeSet[type=boolean, **ranges=1, {[false,false]}] ]}]**
and result.getTupleDomain() = Optional[{c_boolean=[ SortedRangeSet[type=boolean, **ranges=2, {(null,true), (true,null)}] ]}]**

Reply:
okey

Reply:
Let's have a follow up issue to simplify this